### ISSUE TITLE
Error on reading empty worksheet

### ISSUE THREAD

**Issue by msgoussi** [hadley/readxl#222]
Dear Mr.
I am trying to read sheets from xls file

```R
sfile<-"EIU_Data.xls"
sheetlist<-readxl::excel_sheets(sfile)
for (i in 1:length(sheetlist)){
  temp<-readxl::read_excel(sfile,sheet=i,col_names=FALSE)
if (nrow(temp)!=0){
# rest of my own code
}}
```

However I get this error, since the last 3 sheets are empty.
Error in xls_cols(path, sheet, col_names = col_names, col_types = col_types,  : 
  `col_names` and `col_types` must have the same length
Is there a way to solve this problem

[EIU_Data.zip](https://github.com/hadley/readxl/files/675641/EIU_Data.zip)

**Comment by jennybc** [comment]
Last 3 worksheets are boilerplate, empty Sheet1, Sheet2, Sheet3. This is about reading empty worksheets. Same outcome but different error message on xlsx.

``` r
## https://github.com/hadley/readxl/issues/222
library(readxl)
sfile <- "EIU_Data.xls"
sheetlist <- readxl::excel_sheets(sfile)
head(sheetlist)
#> [1] "DGDP" "GDPP" "RGDP" "GDPD" "EXPD" "IMPD"
tail(sheetlist)
#> [1] "INPX"   "INPY"   "EFIR"   "Sheet1" "Sheet2" "Sheet3"
# for (i in 1:length(sheetlist)){
#   temp<-readxl::read_excel(sfile,sheet=i,col_names=FALSE)
#   if (nrow(temp)!=0){
#     # rest of my own code
#   }}
i <- 1
temp <- read_excel(sfile, sheet=i, col_names=FALSE)
str(temp)
#> Classes 'tbl_df', 'tbl' and 'data.frame':    214 obs. of  39 variables:
#>  $ X1 : chr  "No of countries : 202 " NA "Long name" "Percentage change in real GDP, over previous year." ...
#>  $ X2 : chr  "GDP (% real change pa) - DGDP [YQ]" "%" "Note" NA ...
#>  $ X3 : chr  NA NA "Source" "Derived from IMF, World Economic Outlook; Asian Development Bank" ...
#>  $ X4 : chr  NA NA "Eiu Analyst" "Alice Mummery" ...
#> snip, snip

i <- length(sheetlist)
sheetlist[i]
#> [1] "Sheet3"
(temp <- read_excel(sfile, sheet=i, col_names=FALSE))
#> Error in xls_cols(path, sheet, col_names = col_names, col_types = col_types,  : 
#>  `col_names` and `col_types` must have the same length

## an xlsx workbook with a single empty worksheet named Sheet1
df <- read_excel("empty.xlsx")
#>  Error in read_xlsx_(path, sheet, col_names = col_names, col_types = col_types,  : 
#>    Skipped over all data 
```

**Comment by jennybc** [comment]
Make sure to consider the case of header row but no data row (#144, #65).

**Comment by jennybc** [comment]
The xlsx part of this is covered under and fixed by #224.
